### ISSUE TITLE
Troubleshooting Issue

### ISSUE THREAD

**Issue by joe2k1** [artisan-roaster-scope/artisan#1147]
I have the following error message on my Acer Aspire V 14 Touch laptop. The software is not working after an operating system upgrade. I get the following message under Artisan.

Exception : PhidgetManager couldn’t be started. Verify that the Phidget driver is correctly installed. 

I am running 5.15-0.69 generic #76 Ubuntu SMP - March 17 2023. 

Let me know if you need any other information as I would love to get roasting again.

Edit: Yes, it’s 2.8.2 (f525b16) set to work with Phidgets. Again it worked fine before but after the upgrade. After some additional troubleshooting. It seems that the libphidget22 was not installed. I untarred into the downloads folder. I get an error message during the make process. Here is the error below:

rc/vint.h:96:19: note: by argument 4 of type 'const uint8_t *' {aka 'const unsigned char *'} to 'sendVINTDataPacket' declared here
   96 | PhidgetReturnCode sendVINTDataPacket(mosiop_t iop, PhidgetChannelHandle channel,
      |                   ^~~~~~~~~~~~~~~~~~
src/vintpackets.c:516:17: note: 'buf' declared here
  516 |         uint8_t buf[2];
      |                 ^~~
src/vintpackets.c: In function 'sendDCC1000_POSITIONCONTROL_FAILSAFE':
src/vintpackets.c:6520:33: warning: 'buf' may be used uninitialized [-Wmaybe-uninitialized]
 6520 |                         return (sendVINTDataPacket(bp->iop, ch, VINT_PACKET_TYPE_FAILSAFE_RESET, buf, 0));
      |                                ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/phidget.h:219,
                 from src/phidgetbase.h:233,
                 from src/vintpackets.c:1:
src/vint.h:96:19: note: by argument 4 of type 'const uint8_t *' {aka 'const unsigned char *'} to 'sendVINTDataPacket' declared here
   96 | PhidgetReturnCode sendVINTDataPacket(mosiop_t iop, PhidgetChannelHandle channel,
      |                   ^~~~~~~~~~~~~~~~~~
src/vintpackets.c:6428:17: note: 'buf' declared here
 6428 |         uint8_t buf[8];
      |                 ^~~
  CC       src/virtual.lo
  CCLD     [libphidget22.la](http://libphidget22.la/)
ar: `u' modifier ignored since `D' is the default (see `U')
Making phidget22.h
gcc -g -E -P  -I./src -I./src/ext -I./src/ext/include -DPACKAGE_NAME=\"phidget22\ C\ Library\" -DPACKAGE_TARNAME=\"libphidget22\" -DPACKAGE_VERSION=\"1.14.20230331\" -DPACKAGE_STRING=\"phidget22\ C\ Library\ 1.14.20230331\" -DPACKAGE_BUGREPORT=\"[support@phidgets.com](mailto:support@phidgets.com)\" -DPACKAGE_URL=\"[www.phidgets.com](http://www.phidgets.com/)\" -DPACKAGE=\"libphidget22\" -DVERSION=\"1.14.20230331\" -DP22_LIB_VERSION=\"1.14.20230331\" -D_HOST_CPU=x86_64 -D_MACHINE=x86_64 -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DMOS_BYTE_ORDER=MOS_LITTLE_ENDIAN -D_MOS_PTHREADS=1 -DUNIX=1 -D_LINUX=1 -DLinux=1 -DLIBUSB_ASYNC=1 -DNDEBUG=1 -DTHREAD_LOCALE_LOCALE_H=1 -DEXTERNALPROTO ./src/phidget22int.h | sed -e "s/extern \"C\" //g" >> phidget22.h

Unfortunately during the install 

Thanks in advance.

**Comment by CarefreeBuzzBuzz** [comment]
You might have to go into the Security settings and approve the software.  But please update to the latest version.  Save your settings under the Help menu first.

**Comment by joe2k1** [comment]
> You might have to go into the Security settings and approve the software. But please update to the latest version. Save your settings under the Help menu first.
> […](#)

Hi, Thanks for your help. I can confirm that the software is running the latest version. Can you be more specific about security settings and approving the software. It is very clear that something is wrong with compiling the Phidget driver.

**Comment by MAKOMO** [comment]
This is a very confusing report. You edited your initial message such that the replies to it do not make sense anymore.

Obviously you have a problem with your Phidget driver. This is not part of Artisan. Contact Phidget Support on how to proper install their driver on your machine. Without that driver correctly installed Artisan will not be able to connectto your Phidgets.

**Comment by joe2k1** [comment]
Just as a followup, everyone was correct it was an issue of the Phidget driver/library not being installed at all. I am running Linux Mint. I should have mentioned that part earlier. I worked with the technical support department to determine that i had to run these 2 very specific command lines to get Artisan working again. 

curl -fsSL https://www.phidgets.com/downloads/setup_linux | sudo -E bash -

sudo apt-get install -y libphidget22

Once I ran those two lines separately, I was able to get Aristan working again, no more error messages. 

Thanks again to everyone and I am leaving this here to help others who may run into this very strange off issue after an operating system upgrade.
